mantidproject/mantid
# Add new pearl focus mode allowing user to specify which transverse detector modules to focus

### Description of work

New focus mode "trans_custom" added to the PEARL reduction routines which allows user to specify specific modules to include in the transverse bank focussing using the parameter `trans_mod_nums`. The module numbers in the range 1-9 can be specified using the same string syntax as run-numbers - e.g. `trans_mod_nums="1-3,5"` corresponds to focussing modules 1,2,3 and 5.

Fixes #38340

**Report to:** Nick Funnell (PEARL, ISIS)

### To test:

Use the configs in this folder \\olympic\Babylon5\Public\RWaite\PEARLExampleForSarah.zip

(1) Run this script
```
from isis_powder.pearl import Pearl

pearl = Pearl(user_name="Danny", vanadium_normalisation=False, 
                    output_directory=r"C:\Users\xhg73778\Desktop\PEARLExampleForSarah",   #directory above cycle number 
                    config_file=r"C:\Users\xhg73778\Desktop\PEARLExampleForSarah\User_experiment.yaml",
                    calibration_directory=r"C:\Users\xhg73778\Desktop\PEARLExampleForSarah\calib_critical_files",
                    calibration_mapping_file=r"C:\Users\xhg73778\Desktop\PEARLExampleForSarah\calib_critical_files\pearl_run_mapping.yaml",
                    incl_file_ext_in_wsname=True)


pearl.focus(run_number="116374", file_ext=".raw", perform_attenuation=False, 
                  subtract_empty_instrument=False, focus_mode="trans_custom", trans_mod_nums="1-3,5")
```

it should produce these workspaces

![image](https://github.com/user-attachments/assets/10eb61fb-0ae1-45bd-8aa7-f7c4a23b9288)


### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
